packages/components/src/unit-control/index.js

```diff
@@ -159,11 +159,18 @@ function UnitControl(
 		/>
 	) : null;
 
-	// Get current unit type
-	const activeUnit = units.find( ( option ) => option.value === unit );
-
-	// Extract the step
-	const unitStep = activeUnit?.step ? activeUnit.step : 1;
+	let step = props.step;
+	// if no step prop has been passed
+	if ( ! step ) {
+		// Get current unit type
+		let activeUnit = units.find( ( option ) => option.value === unit );
+		// if unit wasn't found, choose the first option
+		if ( ! activeUnit && units.length > 0 ) {
+			activeUnit = units[ 0 ];
+		}
+		// Extract the step if it exists, or default to `1`
+		step = activeUnit?.step ? activeUnit.step : 1;
```

Comment:
Now with the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) we can use:

```suggestion
		step = activeUnit?.step ?? 1;
```

Reply:
Awesome, haven't got these on reflex yet ^^